lib/view_model/after_auth_view_models/add_post_view_models/add_post_view_model.dart

```diff
@@ -154,4 +164,121 @@ class AddPostViewModel extends BaseModel {
     _imageFile = null;
     notifyListeners();
   }
+
+  /// This function will return the start and end index of the current tag where the pointer cursor is currently placed.
+  ///
+  /// **params**:
+  ///   None
+  ///
+  /// **returns**:
+  /// * `MapEntry<int, int>`: Getting the map.
+  MapEntry<int, int> _getCurrentString() {
+    final text = _controller.text;
+    final cursorPosition = _controller.selection;
+    final cursorIndex = cursorPosition.baseOffset - 1;
+
+    int left = 0;
+    int right = 0;
+
+    int temp = cursorIndex;
+    while (temp > 0) {
+      if (text[temp] == ' ' || text[temp] == '\n') {
+        left = temp + 1;
+        break;
+      }
+      temp--;
+    }
+    if (temp == 0) {
+      left = 0;
+    }
+    temp = cursorIndex;
+
+    while (temp < text.length) {
+      if (text[temp] == ' ' || text[temp] == '\n') {
+        right = temp;
+        break;
+      }
+      temp++;
+    }
+    if (temp == text.length) {
+      right = text.length;
+    }
+    return MapEntry(left, right);
+  }
+
+  /// This function is used to fetch the hashtags from server based on the prefix.
+  ///
+  /// **params**:
+  /// * `prefix`: Taking the prefix of the tag.
+  ///
+  /// **returns**:
+  /// * `List<String>`: Answer of life.
+  List<String> _fetchTags(String prefix) {
+    // function that will make a request for trending tags and add it to provide user
+    return ["React", "Flutter", "Dart", "Talawa"];
+  }
+
+  /// This function is used to add the tag in the post(textField) on tapping the one of the tag from suggestions and is responsible to reposition the cursor at the end of the string.
+  ///
+  /// Other description.
+  ///
+  /// **params**:
+  /// * `tag`: The tag that is being added to the string.
+  ///
+  /// **returns**:
+  ///   None
+  void onTagClick(String tag) {
+    final list = _controller.text.split(' ');
+
+    final vals = _getCurrentString();
+    final currString = _controller.text.substring(vals.key, vals.value);
+
+    final index = list.indexOf(currString);
+
+    if (index != -1) {
+      list[index] = "#$tag";
+    } else {
+      list.add("#$tag");
+    }
+
+    _controller.text = list.join(' ');
+
+    _controller.selection = TextSelection.fromPosition(
+      TextPosition(
+        offset:
+            (index == -1) ? _controller.text.length : vals.key + tag.length + 1,
+      ),
+    );
+
+    _showHashtagList = false;
+
+    notifyListeners();
+  }
+
+  /// This function is responsible to detect wether the tag is being added to the string currently and is responsible to reposition the cursor at the end of the string.
+  ///
+  /// Other description.
+  ///
+  /// **params**:
+  /// * `_`: Not to be used
+  ///
+  /// **returns**:
+  ///   None
+  void handleTextChange(_) {
```

Comment:
Can this be better? Maybe some caching and cooldown? Because if there's a lot of text and the user keeps typing, this function will iterate over all of the text unnecessarily.